src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1012,9 +1013,19 @@ def construct_payload(self):
         self.set_up_service_bindings()
         self.set_up_unbind_service_bindings()
         self.set_up_source()
+        self.set_up_containers_for_private_registry_image()
         if self.get_argument_max_inactive_revisions() is not None:
             safe_set(self.new_containerapp, "properties", "configuration", "maxInactiveRevisions", value=self.get_argument_max_inactive_revisions())
 
+    def set_up_containers_for_private_registry_image(self):
+        containers = safe_get(self.new_containerapp, "properties", "template", "containers", default=[])
+        # Check for private registry image and if found remove the containers from the container app and ensure there is only one container with the build image.
+        for container in containers:
```

Comment:
The tricky part is that in that case, we actually want to remove all previous containers, not only the non-Cloud Build ones.
We also can't rely on the use of the default container registry or not, because soon we will have support for BYO ACR and they can be stored anywhere.

I remember that you mentioned that there was an update/append option in the CLI. Could you please share where this is? This would seem to me like the right approach.
The problem is that here we're already at the containerapp_decorator level, so we lost all information about whether we're in a Cloud Build context or not.

Reply:
It may be better to remove all previous containers in `containerapp_up` [here](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/custom.py#L1193) since we determine if cloud build scenario or not here. 

[This](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/containerapp_decorator.py#L305) is the code flow which determines if new container must be appended or replaced, however this flow is also in the containerapp_decorator class so we would not have any information about Cloud Build or not. So it may be better to remove all containers even before updating the container app.